frontend/src/Editor/QueryManager/QueryEditors/Restapi/index.jsx

```diff
@@ -39,11 +39,26 @@ class Restapi extends React.Component {
           this.addNewKeyValuePair('body');
         }
       }, 1000);
+      setTimeout(() => {
+        this.initizalizeRetryNetworkErrorsToggle();
+      }, 1000);
     } catch (error) {
       console.log(error);
     }
   }
 
+  initizalizeRetryNetworkErrorsToggle = () => {
```

Comment:
In my opinion, a better solution would be to take the default value selected  from DataSource at the time of creation if the data query itself.
(https://github.com/ToolJet/ToolJet/pull/10100/files#r1658598402)

Reply:
Actually the default as `null` is being inferred from the [constants.js](https://github.com/ToolJet/ToolJet/pull/10219/files#diff-f538b520c9602ff219c87181df4be1406128921926c195bba66488b0cf5c46a0R86) file itself. We can modify the condition to alter the generic flow, but I think we can stay away from for this one-off case until similar patterns emerge. Wdyt?